encoder.go

```diff
@@ -34,86 +35,57 @@ func maxPow(b uint64) (d uint64, n int) {
 // Encode encodes uuid.UUID into a string using the most significant bits (MSB)
 // first according to the alphabet.
 func (e encoder) Encode(u uuid.UUID) string {
-	if e.alphabet.singleBytes {
-		return e.encodeSingleBytes(u)
-	}
-	return e.encode(u)
-}
-
-func (e encoder) encodeSingleBytes(u uuid.UUID) string {
 	num := uint128{
 		binary.BigEndian.Uint64(u[8:]),
 		binary.BigEndian.Uint64(u[:8]),
 	}
-	var r uint64
+	if e.alphabet.len == defaultBase && e.alphabet.maxBytes == 1 {
+		return e.defaultEncode(num)
+	}
+	return e.encode(num)
+}
+
+func (e encoder) defaultEncode(num uint128) string { // compiler optimizes a lot of divisions by constant
 	var i int
-	var buf []byte
-	if e.alphabet.len == defaultBase { // compiler optimizations using constants for default base
-		buf = make([]byte, defaultEncLen)
-		for i = defaultEncLen - 1; num.Hi > 0 || num.Lo > 0; {
-			num, r = num.quoRem64(defaultDivisor)
-			for j := 0; j < defaultNDigits && i >= 0; j++ {
-				buf[i] = byte(e.alphabet.chars[r%defaultBase])
-				r /= defaultBase
-				i--
-			}
-		}
-	} else {
-		buf = make([]byte, e.alphabet.encLen)
-		l := uint64(e.alphabet.len)
-		d, n := maxPow(l)
-		for i = int(e.alphabet.encLen - 1); num.Hi > 0 || num.Lo > 0; {
-			num, r = num.quoRem64(d)
-			for j := 0; j < n && i >= 0; j++ {
-				buf[i] = byte(e.alphabet.chars[r%l])
-				r /= l
-				i--
-			}
+	var r uint64
+	buf := make([]byte, defaultEncLen)
```

Comment:
Would it make sense to use an array here instead?

```suggestion
	var buf [defaultEncLen]byte
```

Reply:
Yeah, you are right, updated.